testsuites/CBLTester/upgrade_tests/test_cbl_upgrade.py

```diff
@@ -0,0 +1,133 @@
+                        "{} doesn't have encrypted db upgrade support".format(base_liteserv_version))
+        db_config = db.configure(password=db_password)
+        db_prefix = "travel-sample-encrypted"
+    else:
+        db_config = db.configure()
+        db_prefix = "travel-sample"
+
+    temp_db = db.create("temp_db", db_config)
+    time.sleep(1)
+    new_db_path = db.getPath(temp_db)
+    delimiter = "/"
+    if liteserv_platform == "net-msft" or liteserv_platform == "net-uwp":
+        delimiter = "\\"
+    new_db_path = "{}".format(delimiter).join(new_db_path.split(delimiter)[:-2]) +\
+                  "{}{}.cblite2".format(delimiter, upgrade_cbl_db_name)
+    db.deleteDB(temp_db)
+
+    old_liteserv_db_name = ""
+    if base_liteserv_version in supported_base_liteserv:
+        old_liteserv_db_name = db_prefix + "-" + base_liteserv_version
+    else:
+        pytest.skip("Run test with one of supported base liteserv version - ".format(supported_base_liteserv))
+
+    if liteserv_platform == "android":
+        prebuilt_db_path = "/assets/{}.cblite2.zip".format(old_liteserv_db_name)
+    elif liteserv_platform == "xamarin-android":
+        prebuilt_db_path = "{}.cblite2.zip".format(old_liteserv_db_name)
+    else:
+        prebuilt_db_path = "Databases/{}.cblite2".format(old_liteserv_db_name)
+
+    log_info("Copying db of CBL-{} to CBL-{}".format(base_liteserv_version, upgraded_liteserv_version))
+    prebuilt_db_path = db.get_pre_built_db(prebuilt_db_path)
+    assert "Copied" == utils_obj.copy_files(prebuilt_db_path, new_db_path)
+    cbl_db = db.create(upgrade_cbl_db_name, db_config)
+    assert isinstance(cbl_db, MemoryPointer("value")), "Failed to migrate db from previous version of CBL"
+    cbl_doc_ids = db.getDocIds(cbl_db, limit=40000)
+    assert len(cbl_doc_ids) == 31591
+
+    # Replicating docs to CBS
+    sg_client = MobileRestClient()
+    replicator = Replication(base_url)
+    username = "autotest"
+    password = "password"
+
+    # Reset cluster to ensure no data in system
+    c = Cluster(config=cluster_config)
+    c.reset(sg_config_path=sg_config)
+
+    sg_client.create_user(sg_admin_url, sg_db, username, password)
+    authenticator = Authenticator(base_url)
+    cookie, session_id = sg_client.create_session(sg_admin_url, sg_db, username)
+    replicator_authenticator = authenticator.authentication(session_id, cookie, authentication_type="session")
+    repl_config = replicator.configure(cbl_db, sg_blip_url, replication_type="push",
+                                       replicator_authenticator=replicator_authenticator)
+    repl = replicator.create(repl_config)
+    replicator.start(repl)
+    replicator.wait_until_replicator_idle(repl, sleep_time=10)
+    total = replicator.getTotal(repl)
+    completed = replicator.getCompleted(repl)
+    assert total == completed
+    log_info("total:", total)
+    log_info("completed:", completed)
+    replicator.stop(repl)
+
```

Comment:
can we add verification whether all docs got replicated?
Also looks like we need to add few more steps like update docs, share docs, delete docs, add more docs . All basic operations done manually. After update,

Reply:
Added